### ISSUE TITLE
Bug: Bad authorization state. Refreshing the page might solve the issue. (

### ISSUE THREAD

**Issue by pablojakub** [soofstad/react-oauth2-pkce#85]
Hello, 
In my react React I've got an error when I first time enter the page: 
![image](https://user-images.githubusercontent.com/68683285/234872710-4639cfae-54b6-4036-8fbe-2c8e6cd40a1f.png)

![image](https://user-images.githubusercontent.com/68683285/234872818-adfd83fa-7c18-4c9b-8d77-b0542777bdcf.png)
THis may indicates that in my url there is no 'code' query. Indeed there is no 'code' query and after reloading the page it is. 

I tried to change settings in my router: 
![image](https://user-images.githubusercontent.com/68683285/234873121-d5f6c9f0-c77b-43ed-b643-f3400e7a2b1f.png)
Right now default page redirects to /dashboard. But even when I delete it doesn't work. 

In TAuthConfig I have autoLogin: true and clearURL: false,

I'am using this verson: 
![image](https://user-images.githubusercontent.com/68683285/234873563-e60e25b8-30ef-41db-9747-591be4b2e7e1.png)

My expected behaviour - login wihout refreshing the page. Of course I can force login() method but it does not do the trick for me in this case

**Comment by soofstad** [comment]
Could you be running into this issue? https://github.com/soofstad/react-oauth2-pkce#after-redirect-back-from-auth-provider-with-code-no-token-request-is-made

**Comment by pablojakub** [comment]
Yeah I read about it and it's not this case. When I'm redirected from my provider everything works fine. It happens when I enter the page -> provider/ library somehow remembers session -> no redirect to provider to login -> error 'bad authorization state...'

All my routes are wrapper inside AuthProvider. AuthProvider is in index.sx and App routes in App.tsx one level lower

**Comment by soofstad** [comment]
Are you able to make minimal example on how to recreate this bug?

If not, I'd like so see the state of localStorage, both before and after you are redirected to the authentication provider. If everything there looks alright, then you should have a look at what parameters the auth servers sets when it redirects you back after login.

**Comment by pablojakub** [comment]
Ok - most updated version:

First scenario: 
1) Local storage is empty - everything works fine. I'm redirected to provider -> there log in -> redirect to my app and everything works

Second scenario:
Local storage is NOT empty. There is some ROCP data. I'm NOT redirected to provider -> there is no 'code' param in my url and error occurs. 
My local storage: 
![image](https://user-images.githubusercontent.com/68683285/235091813-5332f09d-943a-4607-a07b-5839884b7124.png)

I think in order to reproduce it I need to: 
1) Log in to the provider first OUTSIDE my application environment. 
2) Turn on my app
3) Then there is no redirect and beforementioned data in local storage.

**Comment by soofstad** [comment]
`react-oauth2-code-pkce` will only attempts to retrive `code` from url if `loginInProgress` is true.
That is set right before redirecting to auth provider.
The login flow needs to start from the web app using the package. Any redirects to the web-app with code besides that will not work. However, it should just automatically log the user in anyway, with a new redirect.

Have you tried clearing all the web apps persistent data? Calling "logout()" should be enough.

If you wan't some more help with this, I realy need an example on how to recreate it.
